pliers/config.py

```diff
@@ -1,10 +1,88 @@
-cache_transformers = True
-log_transformations = True
-drop_bad_extractor_results = True
-progress_bar = True
-parallelize = False
-n_jobs = None
-default_converters = {
-    'AudioStim->TextStim': ('IBMSpeechAPIConverter', 'WitTranscriptionConverter'),
-    'ImageStim->TextStim': ('GoogleVisionAPITextConverter', 'TesseractConverter')
+''' The config module contains package-level settings and tools for
+manipulating them. '''
+
+import json
+from os.path import join, expanduser, exists
+import os
+from six import string_types
+
+__all__ = ['set_option', 'set_options', 'get_option']
+
+_config_name = 'pliers_config.json'
+
+_default_converters = {
+    'AudioStim->TextStim':
+        ('IBMSpeechAPIConverter', 'WitTranscriptionConverter'),
+    'ImageStim->TextStim':
+        ('GoogleVisionAPITextConverter', 'TesseractConverter')
 }
+
+_default_settings = {
+    'cache_transformers': True,
+    'default_converters': _default_converters,
+    'drop_bad_extractor_results': True,
+    'log_transformations': True,
+    'n_jobs': None,
+    'parallelize': False,
+    'progress_bar': True,
+    'use_generators': False
+}
+
+
+def set_option(key, value):
+    if key not in _settings:
+        raise ValueError("Invalid pliers setting: '%s'" % key)
+    _settings[key] = value
+
+
+def set_options(**kwargs):
+    for k, v in kwargs.items():
+        set_option(k, v)
+
+
+def get_option(key):
+    if key not in _settings:
+        raise ValueError("Invalid pliers setting: '%s'" % key)
+    return _settings[key]
+
+
+def from_file(filenames, error_on_missing=True):
+    if isinstance(filenames, string_types):
+        filenames = [filenames]
+    for f in filenames:
+        if exists(f):
+            settings = json.load(open(f))
+            _settings.update(settings)
+        elif error_on_missing:
+            raise ValueError("Config file '%s' does not exist." % f)
```

Comment:
Would also be useful to have a method to save the current config settings (default to the home directory standard?). Not high priority